pkg/werf/exec/cancel.go

```diff
@@ -27,6 +29,12 @@ func CommandContextCancellation(ctx context.Context, name string, arg ...string)
 	return PrepareGracefulCancellation(exec.CommandContext(ctx, name, arg...))
 }
 
+func TerminateIfCanceled(ctx context.Context, err error, exitCode int) {
```

Comment:
Looks nice! Let's simplify code a little bit: place  deriving of exit_code to under the hood of this function.